packages/ng-datasource/src/progress.data.ng.ds.js

```diff
@@ -108,7 +108,8 @@ var DataSource = /** @class */ (function () {
         }
         if (this.readLocal && this._initFromServer) {
             return Observable_1.Observable.create(function (observer) {
-                observer.next(_this.getJsdoData());
+                var data = _this.getJsdoData();
+                observer.next({ data: data, total: data.length });
             });
         }
         if (params) {
```

Comment:
Hi Maura,
Do we have a scenario where we want to read data from server in this block of code. If so, we might want to change the total property to **data.total** instead of **data.length**
Thanks,
Anil Kumar

Reply:
No. readLocal = true and  the parent dataSource has already retreived data from the server (since _initFromServer is true), so this is always returning the local data.
As you can see, there is no call to the BE for data